### PR TITLE
[SMALL FIX] Remove a precondition check msg

### PR DESCRIPTION
**Pull request by peisun1115**
@gpang

### CODE REVIEW

**Comment by gpang** [comment]
Should this just be a precondition message constant? Does this significantly affect performance?

**Comment by peisun1115** [comment]
I've tested it. the impact is not big. Personally, i think it is an overkill here to add a constant for this..  I don't think the msg can help debugging much. But I can add it if you prefer.

File: underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystemFactory.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public HdfsUnderFileSystemFactory() {}
 
   @Override
   public UnderFileSystem create(String path, Configuration configuration, Object conf) {
-    Preconditions.checkArgument(path != null, "path may not be null");
+    Preconditions.checkArgument(path != null);
```
Comment (aaudiber) [inline underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystemFactory.java]: How about `Preconditions.checkNotNull(path, "cannot create a file system for a null path")`?

Comment (apc999) [inline underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystemFactory.java]: put `"cannot create a file system for a null path"` into a constant in `PreconditionMessage`?
It is still friendly if we could have an error message

**Comment by apc999** [comment]
left a comment

**Comment by calvinjia** [comment]
@peisun1115 If we just make it a constant, does this PR still serve the original purpose? The string constant should have the same performance as the raw string.

**Comment by apc999** [comment]
@calvinjia yes, it seems to me that there should be no performance difference between referencing a string literal (original way) and referencing a string constant (what I and @aaudiber proposed). Now I recall that, it seems we only migrate string literals to `PreconditionMessage` when we have unit test covering this argument check, which is not true here.

**Comment by peisun1115** [comment]
@apc999 @calvinjia Lets' abandon this PR. This part of logic will be removed in my next PR. We can continue the discussion about the precondition msg though. I am not quite sure about the performance impact of the precondition msg.

**Pull request closed without merging**